twenty-first/benches/extrapolation.rs

```diff
@@ -0,0 +1,218 @@
+use criterion::criterion_group;
+use criterion::criterion_main;
+use criterion::BenchmarkId;
+use criterion::Criterion;
+
+use itertools::Itertools;
+use num_traits::One;
+use twenty_first::math::ntt::intt;
+use twenty_first::math::other::random_elements;
+use twenty_first::math::traits::FiniteField;
+use twenty_first::math::traits::PrimitiveRootOfUnity;
+use twenty_first::math::zerofier_tree::ZerofierTree;
+use twenty_first::prelude::*;
+
+criterion_main!(benches);
+criterion_group!(
+    name = benches;
+    config = Criterion::default().sample_size(10);
+    targets = extrapolation<{ 1 << 18 }, { 1 << 6 }>,
+              extrapolation<{ 1 << 18 }, { 1 << 7 }>,
+              extrapolation<{ 1 << 18 }, { 1 << 8 }>,
+              extrapolation<{ 1 << 19 }, { 1 << 6 }>,
+              extrapolation<{ 1 << 19 }, { 1 << 7 }>,
+              extrapolation<{ 1 << 19 }, { 1 << 8 }>,
+              extrapolation<{ 1 << 20 }, { 1 << 6 }>,
+              extrapolation<{ 1 << 20 }, { 1 << 7 }>,
+              extrapolation<{ 1 << 20 }, { 1 << 8 }>,
+);
+
+fn intt_then_evaluate(
+    codeword: &[BFieldElement],
+    offset: BFieldElement,
+    zerofier_tree: &ZerofierTree<BFieldElement>,
+    shift_coefficients: &[BFieldElement],
+    tail_length: usize,
+) -> Vec<BFieldElement> {
+    let omega = BFieldElement::primitive_root_of_unity(codeword.len() as u64).unwrap();
+    let log_domain_length = codeword.len().ilog2();
+    let mut coefficients = codeword.to_vec();
+    intt(&mut coefficients, omega, log_domain_length);
+    let polynomial: Polynomial<BFieldElement> = Polynomial::new(coefficients)
+        .scale(offset.inverse())
+        .reduce_by_ntt_friendly_modulus(shift_coefficients, tail_length);
+    polynomial.divide_and_conquer_batch_evaluate(zerofier_tree)
+}
+
+#[allow(dead_code)]
```

Comment:
Explained in 5a9b78d9fdf25ea722358bd09d7887cc9b1a6ed0:
```
// The benchmark for iterative barycentric is currently commented-out because
// slow.
```